### ISSUE TITLE
Upgraded to version 3.3.0 - LT14 fails within WINDOW FUNCTION

### ISSUE THREAD

**Issue by moseleyi** [sqlfluff/sqlfluff#6495]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The ruls checks for keywords to be in new line but forgets the context of a window function.

### Expected Behaviour

```
SELECT
  DENSE_RANK() OVER (ORDER BY id)
```

Passes the rule

### Observed Behaviour

Error is reported
```The 'ORDER' keyword should always start a new line.```

### How to reproduce

Make sure you upgrade sqlfluff to 3.3.0

### Dialect

bigquery

### Version

3.3.0

### Configuration

```
[sqlfluff]
templater = dbt
dialect = postgres
exclude_rules = RF02, AL01, AL05, ST07, ST06, LT05

; LT05 - Lines are too long ( > 80) it's okay
; ST06 - Select wildcards then simple targets before calculations - unfortunately it applies to window functions like DENSE_RANK

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = true
indented_using_on = false

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:aliasing.length]
min_alias_length = 3
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
This is working as intended, but I agree it's a slightly confusing result. In the context of a longer window function it makes more sense:
```sql
select
    rank(my_column) ignore nulls over (partition by a, b, c order by d, e, f rows between 42 preceding and 79 following)
from foo
```

This would get coerced to:

```sql
select
    rank(my_column) ignore nulls over (
        partition by a, b, c
        order by d, e, f
        rows between 42 preceding and 79 following
    )
from foo
```

@keraion @WittierDinosaur  - do either of you have views on how we should change the defaults given this example? I'll have a toy with the configs myself.

@moseleyi - this behaviour is entirely controlled by the `[sqlfluff:layout:type:orderby_clause]` config section. I'm totally up for suggestions on how we might modify the default config to make this behaviour more in line with expectations.

**Comment by moseleyi** [comment]
I don't split window functions in multiple lines - I was pondering that but in the end it's not a function that has multiple arguments split by comma. Also ROWS usually can be used as "WINDOW" keyword at the end. Therefore window functions are rather shorter than longer - purely assumption here. But either way. Any keyword in the window function shouldn't be treated the same as part of the whole query

**Comment by mroy-seedbox** [comment]
The same also happens with aggregate/window functions which only support one single `ORDER BY` statement in a `WITHIN GROUP` clause (e.g. [`LISTAGG` in Snowflake](https://docs.snowflake.com/en/sql-reference/functions/listagg)).

For example: `LISTAGG(col, ',') WITHIN GROUP (ORDER BY col)`

**Comment by johalnes** [comment]
I'm totally in favor of this rule as a whole, but meets the same issues as @moseleyi. I'm also not getting the same nice formating as you did @alanmcruickshank, sqlfmt formats my window function with order by like this: 
```sql
select
    *,
    coalesce(
        lead(tc_valid_from_date) over (
            partition by agr_line_no, agr_line_seq_no order by tc_valid_from_date
        ),
        '9999-12-31'
    ) as tc_valid_to_date
from object_max
```
After running sqlfluff --fix I get the following, where it seems to assume it's an ordinary order by:
```sql
select
    *,
    coalesce(
        lead(tc_valid_from_date) over (
            partition by agr_line_no, agr_line_seq_no
order by tc_valid_from_date
        ),
        '9999-12-31'
    ) as tc_valid_to_date
from
    object_max
```

Would use the order by config for now, and would love to have an option to skip window functions instead of having to disable order by in the future 😄

**Comment by johalnes** [comment]
FYI - I get same kind of error and fixing when using where not exist logic (but not sure if this is considered anti-pattern or not):
```sql
        select *
        from source
        where
            not exists (
                select 1 from all_x where source.policy_seq_no = all_x.policy_seq_no
            )
```
becomes 
```sql
        select *
        from source
        where
            not exists (
                select 1 from all_x
        where source.policy_seq_no = all_x.policy_seq_no
            )
```

**Comment by mfvanek** [comment]
One more case for PostgreSQL

```sql
        select
            constraint_name,
            table_oid,
            foreign_table_oid,
            array_agg(attname::text || ',' || attnotnull::text order by attposition) as columns // I do not want a new line here
        from fk_with_attributes
        group by constraint_name, table_oid, foreign_table_oid
```
